File: src/main/java/me/jameschan/burrow/command/builtin/ExistCommand.java
package me.jameschan.burrow.command.builtin;

import me.jameschan.burrow.command.Command;
import me.jameschan.burrow.context.RequestContext;
import picocli.CommandLine;

@CommandLine.Command(
    name = "exist",
    mixinStandardHelpOptions = true,
    description = "Check if an entry exist."
)
public class ExistCommand extends Command {
    @CommandLine.Parameters(index = "0")
    private Integer id;

    public ExistCommand(RequestContext context) {
        super(context);
    }

    @Override
    public Integer call() {
        final var hoard = context.getHoard();
        buffer.append(hoard.exist(id) ? "true" : "false");

        return 0;
    }
}

File: src/main/kotlin/carton/hoard/command/ExistCommand.kt
package burrow.carton.hoard.command

import burrow.carton.hoard.Hoard
import burrow.kernel.terminal.*

@BurrowCommand(
    name = "exist",
    header = ["Checks if an entry exists."]
)
class ExistCommand(data: CommandData) : Command(data) {
    @Parameters(index = "0", description = ["The ID of the entry."])
    private var id: Int = 0

    override fun call(): Int {
        if (id <= 0) {
            stderr.println(
                "Error: Entry ID must be a positive integer. Provided ID: $id"
            )
            return ExitCode.USAGE
        }

        val isExist = use(Hoard::class).exists(id)
        stdout.println(if (isExist) "true" else "false")

        return ExitCode.OK
    }
}